cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/RightsizeRequest.java

```diff
@@ -42,26 +43,63 @@ protected RightsizeResult handle() {
-                                                       String.join(" ,", topicNamesMatchedWithPattern)));
-    } else {
-      topicName = topicNamesMatchedWithPattern.iterator().next();
-    }
-    ProvisionRecommendation recommendation =
-        new ProvisionRecommendation.Builder(ProvisionStatus.UNDER_PROVISIONED).numBrokers(_parameters.numBrokersToAdd())
-                                                                              .numPartitions(_parameters.partitionCount())
-                                                                              .topic(topicName)
-                                                                              .build();
+    ProvisionRecommendation recommendation = sanityCheckResources();
     Map<String, ProvisionRecommendation> provisionRecommendation;
     provisionRecommendation = Collections.singletonMap(RECOMMENDER_UP, recommendation);
 
     ProvisionerState provisionerState = provisioner.rightsize(provisionRecommendation, new RightsizeOptions());
 
-    return new RightsizeResult(_parameters.numBrokersToAdd(), _parameters.partitionCount(), topicName, provisionerState, config);
+    return new RightsizeResult(_parameters.numBrokersToAdd(), _parameters.partitionCount(), _topicName, provisionerState, config);
+  }
+
+  /**
+   * Ensure that
+   * <ul>
+   *   <li>exactly one resource type is set</li>
+   *   <li>if the resource type is partition, then the corresponding topic must be specified; otherwise, the topic cannot be specified</li>
+   *   <li>if the resource type is partition, then only one topic must be specified</li>
+   * </ul>
+   *
+   * @return The {@link ProvisionRecommendation} to recommend for rightsizing
+   * @throws IllegalArgumentException when the sanity check fails.
+   */
+  private ProvisionRecommendation sanityCheckResources() throws IllegalArgumentException {
+    ProvisionRecommendation recommendation;
+
+    // Validate multiple resources are not set
+    if (_parameters.numBrokersToAdd() != ProvisionRecommendation.DEFAULT_OPTIONAL_INT
+        && _parameters.partitionCount() == ProvisionRecommendation.DEFAULT_OPTIONAL_INT) {
+      //Validate the topic cannot be specified when the resource type is not partition.
+      if (_parameters.topic() != null) {
+        throw new IllegalArgumentException("When the resource type is not partition, topic cannot be specified.");
+      }
+      recommendation = new ProvisionRecommendation.Builder(ProvisionStatus.UNDER_PROVISIONED).numBrokers(_parameters.numBrokersToAdd())
+                                                                                             .build();
+    } else if (_parameters.numBrokersToAdd() == ProvisionRecommendation.DEFAULT_OPTIONAL_INT
+               && _parameters.partitionCount() != ProvisionRecommendation.DEFAULT_OPTIONAL_INT) {
+      // Validate the topic parameter is not null
+      if (_parameters.topic() == null) {
```

Comment:
Should we also check whether the topic is empty?

Reply:
Here, the `_parameter.topic()` is the Pattern from the parameter so a null check alone is okay. Instead, adding a check to see if the topic is empty against `topicNamesMatchedWithPattern.iterator().next()`.

Reply:
Skipping empty check for the pattern makes sense. I noticed that if a topic is found in the cluster, it cannot be empty. So I don't think any change is needed here.